### PR TITLE
Dx 107 implement the page

### PR DESCRIPTION
**Pull request by QuentinMllt**

### CODE REVIEW

File: samples/dsm/akeneo-dsm/back-end/tests/use-cases/api/product-retrieve.test.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ const { doPimProductsRetrieve } = require('../../../use-cases/api/product-retrie
         await pimProductsRetrieve(req, res, next);
 
         expect(res.json).toHaveBeenCalledWith({
-            'no access token': 'Missing access token in database'
+            'access_token': 'Missing access token in database'
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/tests/use-cases/api/product-retrieve.test.js:90]:
```suggestion
            'access_token': 'Missing access token in the database'
```

File: samples/dsm/akeneo-dsm/front-end/src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,45 @@
...
+}
 
 function App() {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [missingAccessToken, setMissingAccessToken] = useState<string | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetch('http://localhost:8081/some-products', { mode: 'cors' })
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/src/App.tsx:23]: Variabilize http://localhost:8081 in .env: they can use another port I guess

File: samples/dsm/akeneo-dsm/front-end/src/component/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    return (
+        <section>
+            <img className='app-logo' src={logo} alt="akeneo custom app logo" />
+            <div className='breadcrumb'>
+                <Breadcrumb>
+                <Breadcrumb.Step>
+                    PRODUCT LIST
+                </Breadcrumb.Step>
+                <Breadcrumb.Step>
+                    <a href={pimInstance ? JSON.parse(pimInstance)['pim-instance'] : '#'}>
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/src/component/Header.tsx:14]: If no pim-instance, you should not have a clickable link

File: samples/dsm/akeneo-dsm/front-end/src/page/Products.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    family: string;
+    categories: string[];
+    enabled: boolean;
+}
+
+function Products({ products, token, loading }: { products: Product[], token: string | null, loading: boolean }) {
+    const [pimInstance, setPimInstance] = useState<string | null>(null); 
+
+    useEffect(() => {
+        fetch('http://localhost:8081/my-pim-instance', { mode: 'cors' })
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/src/page/Products.tsx:17]: Same, http://localhost:8081 in var

File: samples/dsm/akeneo-dsm/front-end/src/page/Products.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                console.log('error : ' + err);
+            });
+    }, []);
+
+    return (
+        <>
+            <Header pimInstance={pimInstance}/>
+            
+            {token ? ( 
+                <p className='missing_token'>You haven’t an App Token yet. Please set one by following your README</p>
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/src/page/Products.tsx:32]:
```suggestion
                <p className='missing_token'>You haven’t an App Token yet. Please set one by following our README</p> 
```

Comment (tomglvng) [reply samples/dsm/akeneo-dsm/front-end/src/page/Products.tsx:32]: + add a link to the README

Comment (QuentinMllt) [reply samples/dsm/akeneo-dsm/front-end/src/page/Products.tsx:32]: I will create it and add the link in the next task.

File: samples/dsm/akeneo-dsm/back-end/app.js [hunk 6]
```diff
@@ -22,6 +22,11 @@ app.use(express.urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
 
+app.use(function(req, res, next) {
+  res.header("Access-Control-Allow-Origin", "*");
```
Comment (JulienVerbrugge) [inline samples/dsm/akeneo-dsm/back-end/app.js:26]: I'm curious to know why do you have to set this header to * ?

File: samples/dsm/akeneo-dsm/front-end/src/component/Table.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
+import { Badge,
+    Table } from 'akeneo-design-system';
+
+interface Product {
```
Comment (JulienVerbrugge) [inline samples/dsm/akeneo-dsm/front-end/src/component/Table.tsx:4]: Why do we have to repeat this object three times ?

File: samples/dsm/akeneo-dsm/back-end/app.js (hunk 6 above)
Comment (QuentinMllt) [reply samples/dsm/akeneo-dsm/back-end/app.js:26]: You're right, there's no need to wide open the app. It will now acccept only localhost:3000 request

**Review by tomglvng — APPROVED** [review]

**Pull request merged**